src/track/serato/beatsimporter.cpp

```diff
@@ -59,20 +59,46 @@ QVector<double> SeratoBeatsImporter::importBeatsAndApplyTimingOffset(
 
         beats.reserve(beats.size() + pMarker->beatsTillNextMarker());
         for (quint32 j = 0; j < pMarker->beatsTillNextMarker(); ++j) {
-            beats.append(streamInfo.getSignalInfo().millis2frames(
-                    beatPositionMillis + (j * beatLengthMillis)));
+            beats.append(streamInfo.getSignalInfo().millis2frames(beatPositionMillis));
```

Comment:
Couldn't we add `timingOffsetMillis` only to the final value here instead of twice in the loop body? We only need it to adjust the final value, not for the relative calculations. Calculation of the terminal marker should then also be modified.

Reply:
Done.